login-module/src/index.js

```diff
@@ -183,25 +158,50 @@ window.GraphAcademyLogin = class GraphAcademyLogin {
 		const certificateElement = $('#cert-result');
 		if (certificateElement.length) {
 			certificateElement.html("<i>... Checking for certificate ...</i>");
-			const [err, result] = await certificate.getClassCertificate(accessToken);
+			const [err, result] = await services.getClassCertificate();
 			if (result && result.data && result.data.url) {
 				$('#cert-result').html("<a href=\"" + result.data['url'] + "\">Download Certificate</a>");
 			} else {
 				$('#cert-result').html("Certificate not available yet.  Did you complete the quizzes at the end of each section?");
 			}
 		}
 	}
+}
+
+window.GraphAcademyLogin = class GraphAcademyLogin {
+	constructor(options = {}) {
+		const defaultOptions = {
+			stage: 'prod',
+			trainingClassName: null,
+			classStates: {},
+			isCourseLandingPage: false,
+			enrollmentUrl: null,
+		}
+		this.options =  { ...defaultOptions, ...options }
+		this.core = new GraphAcademyCore(this.options);
+	}
+
+	async init() {
+		const { options, core } = this;
+		misc.handleHtmlOnState('checkingSession', options);
+		const [err, result] = await core.login();
+		let graphAcademyPage;
+		if (err) {
+			console.error('Unable to authenticate the user');
+		} else {
+			graphAcademyPage = new GraphAcademyPage({ ...options, services: result });
+			await graphAcademyPage.handleEnrollment();
+		}
+		misc.handleHtmlOnState(err ? 'notLoggedIn' : 'loggedIn', options);
+		return [err, graphAcademyPage];
```

Comment:
returns a `GraphAcademyPage` on which you can access to the authenticated services.